src/scripts/main.js

```diff
@@ -1 +1,66 @@
 'use strict';
+
+const notification = document.querySelector('[data-qa="notification"]');
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    clearTimeout(timeoutId);
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+
+  document.addEventListener('click', function handleClick(e) {
+    if (e.button === 0) {
+      document.removeEventListener('click', handleClick);
+      clearTimeout(timeoutId);
+      resolve('First promise was resolved');
+    }
+  });
+});
+
+firstPromise.then((message) => {
+  notification.classList.add('success');
+  notification.textContent = message;
+}).catch((error) => {
+  notification.classList.add('warning');
+  notification.textContent = error.message;
+});
+
+const secondPromise = new Promise((resolve) => {
+  function handleClick(e) {
+    if (e.button === 0 || e.button === 2) {
+      document.removeEventListener('click', handleClick);
+      resolve('Second promise was resolved');
+    }
+  }
+  document.addEventListener('click', handleClick);
+});
+
+secondPromise.then((message) => {
+  notification.classList.add('success');
+  notification.textContent = message;
+});
+
+const thirdPromise = new Promise((resolve) => {
+  const clicks = {
+    left: false, right: false,
+  };
```

Comment:
Let's try to move them to the upper scope and change for other eventlisteners too